MantidPlot/src/WindowFactory.h

```diff
@@ -0,0 +1,203 @@
+#ifndef WINDOWFACTORY_H
+#define WINDOWFACTORY_H
+
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+#include "MantidAPI/DllConfig.h"
+#include "MantidKernel/make_unique.h"
+#include "MantidKernel/Exception.h"
+#include "MantidKernel/SingletonHolder.h"
+#include "Mantid/IProjectSerialisable.h"
+
+#include <cstring>
+#include <map>
+#include <vector>
+
+#ifndef Q_MOC_RUN
+#include <boost/shared_ptr.hpp>
+#endif
+
+//----------------------------------------------------------------------
+// Forward declarations
+//----------------------------------------------------------------------
+class ApplicationWindow;
+
+namespace Mantid {
+namespace API {
+
+/* Cross platform string case insensitive string comparator functor.
+ */
+struct CaseInsensitiveStringComparator {
+  bool operator()(const std::string &s1, const std::string &s2) const {
+#ifdef _MSC_VER
+    return _stricmp(s1.c_str(), s2.c_str()) < 0;
+#else
+    return strcasecmp(s1.c_str(), s2.c_str()) < 0;
+#endif
```

Comment:
* Can this be reused from above (or the other way around)?
* Can you use the `boost::iequals`?